turf/src/commonMain/kotlin/io/github/elcolto/geokjson/turf/booleans/PointOnLine.kt

```diff
@@ -99,9 +99,9 @@ private fun isPointOnLineSegment(
     }
 
     return when (excludeBoundary) {
-        BoundaryExclusion.START -> isOnLine && (x != x1 && y != y1)
-        BoundaryExclusion.END -> isOnLine && (x != x2 && y != y2)
-        BoundaryExclusion.BOTH -> isOnLine && (x != x1 && x != x2 && y != y1 && y != y2)
+        BoundaryExclusion.START -> isOnLine && !(x == x1 && y == y1)
+        BoundaryExclusion.END -> isOnLine && !(x == x2 && y == y2)
+        BoundaryExclusion.BOTH -> isOnLine && !((x == x1 && y == y1) || (x == x2 && y == y2))
```

Comment:
This was hard to spot. The current tested position may not be exactly the same as the start/end but it may be on the same either x or y axis. Tests still go green so i think this should do the trick